src/orion/serving/experiments_resource.py

```diff
@@ -0,0 +1,160 @@
+=====================================================================================
+
+.. module:: experiments_resource
+   :platform: Unix
+   :synopsis: Represent the experiments/ REST endpoint
+
+"""
+import json
+
+from falcon import Request, Response
+import falcon.status_codes
+
+from orion.core.io import experiment_builder
+from orion.core.utils.exceptions import NoConfigurationError
+from orion.core.worker.experiment import Experiment
+from orion.storage.base import get_storage
+
+
+class ExperimentsResource(object):
+    """Handle requests for the experiments/ REST endpoint"""
+
+    def __init__(self):
+        self.storage = get_storage()
+
+    def on_get(self, req: Request, resp: Response):
+        """Handle the GET requests for experiments/"""
+        experiments = self.storage.fetch_experiments({})
+        leaf_experiments = _find_latest_versions(experiments)
+
+        result = []
+        for name, version in leaf_experiments.items():
+            result.append({
+                'name': name,
+                'version': version
+            })
+
+        resp.body = json.dumps(result, indent=4)
+
+    def on_get_experiment(self, req: Request, resp: Response, name: str):
+        """
+        Handle GET requests for experiments/:name where `name` is
+        the user-defined name of the experiment
+        """
+        try:
+            _verify_experiment_parameters(req.params)
+
+            version = int(req.params['version']) if 'version' in req.params else None
+            experiment = experiment_builder.build_view(name, version)
+
+            import pprint
+            pprint.pprint(experiment.stats)
+
+            response = {
+                "name": experiment.name,
+                "version": experiment.version,
+                "status": _retrieve_status(experiment),
+                "trialsCompleted": experiment.stats['trials_completed'] if experiment.stats else 0,
+                "startTime": str(experiment.stats['start_time']) if experiment.stats else None,
+                "endTime": str(experiment.stats['finish_time']) if experiment.stats else None,
+                "user": experiment.metadata['user'],
+                "orionVersion": experiment.metadata['orion_version'],
+                "config": {
+                    "maxTrials": experiment.max_trials,
+                    "poolSize": experiment.pool_size,
+                    "algorithm": _retrieve_algorithm(experiment),
+                    "space": experiment.configuration['space']
+                },
+                "bestTrial": _retrieve_best_trial(experiment)
+            }
+
+            resp.body = json.dumps(response, indent=4)
+
+        except NoConfigurationError:
+            response = {
+                "message": f"Experiment '{name}' does not exist"
+            }
+            resp.body = json.dumps(response, indent=4)
+            resp.status = falcon.HTTP_404
+
+        except ValueError as exception:
```

Comment:
What are the expected ValueError?

Reply:
It is raised by `_verify_experiment_parameters()` if a parameter given in the query string doesn't exist. I agree it's a bit too general to catch here. In this case, do you think it's better overall the raise a custom exception and catch it, or to simply make `_verify_experiment_parameters()` return a bool and return immediately with the error code if a parameter is unsupported.

Reply:
I would favor the bool approach in this case. It's quite local, so custom error would not be very reusable.

Reply:
Sounds good. I agree. I wasn't sure what was the most pythonic way to solve this.